internal/services/extendedlocation/extended_location_custom_location_test.go

```diff
@@ -0,0 +1,352 @@
+  name                = "acctestcustomlocation%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_extension_ids = [
+    "${azurerm_arc_kubernetes_cluster_extension.test.id}"
+  ]
+  display_name     = "customlocation%[2]d"
+  namespace        = "namespace%[2]d"
+  host_resource_id = azurerm_arc_kubernetes_cluster.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func (r CustomLocationResource) update(data acceptance.TestData, credential string, privateKey string, publicKey string) string {
+	template := r.template(data, credential, publicKey, privateKey)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_extended_custom_location" "test" {
+  name                = "acctestcustomlocation%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  cluster_extension_ids = [
+    "${azurerm_arc_kubernetes_cluster_extension.test.id}",
+  ]
+
+  display_name     = "customlocationupdate%[2]d"
+  namespace        = "namespace%[2]d"
+  host_resource_id = azurerm_arc_kubernetes_cluster.test.id
+}
+`, template, data.RandomInteger)
+}
+
+func (r CustomLocationResource) template(data acceptance.TestData, credential string, publicKey string, privateKey string) string {
+	data.Locations.Primary = "eastus"
+	provisionTemplate := r.provisionTemplate(data, credential, privateKey)
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+  }
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "acctestnw-%[1]d"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.test.name
+  virtual_network_name = azurerm_virtual_network.test.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_public_ip" "test" {
+  name                = "acctestpip-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  allocation_method   = "Static"
+}
+
+resource "azurerm_network_interface" "test" {
+  name                = "acctestnic-%[1]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.test.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.test.id
+  }
+}
+
+resource "azurerm_network_security_group" "my_terraform_nsg" {
+  name                = "myNetworkSecurityGroup"
```

Comment:
Could we update this to use a random name?

Reply:
Sure.